### PR TITLE
 [QF-4667] fix: Use user's WBW language preference in Study Mode popup

### PR DESCRIPTION
**Pull request by yousefdergham**
## Summary                                                                                                          
                                                                                                                    
  Fixed an issue where the Word-by-Word (WBW) translation in the Study Mode popup was always showing English instead
  of the user's selected WBW language preference (e.g., Urdu).

  The `wordTranslationLanguage` parameter was hardcoded to `'en'` in the API request. Changed it to use the
  `selectWordByWordLocale` Redux selector to respect user settings.

  Closes: [QF-4667](https://quranfoundation.atlassian.net/browse/QF-4667)

  ## Type of Change

  - [x] 🐛 Bug fix (non-breaking change which fixes an issue)
  - [ ] ✨ New feature (non-breaking change which adds functionality)
  - [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] 📝 Documentation update
  - [ ] ♻️ Refactoring (no functional changes)

  ## Scope Confirmation

  - [x] This PR addresses **one** feature/fix only
  - [x] If multiple changes were needed, they are split into separate PRs

  ## Rollback Safety

  - [x] Can be safely reverted without data issues or migrations
  - [ ] Rollback requires special steps (describe below):

  ## Test Plan

  - [ ] Unit tests added/updated
  - [ ] Integration tests added/updated
  - [x] Manual testing performed

  **Testing steps:**

  1. Go to Settings and set WBW language to Urdu (or any non-English language)
  2. Open any Surah
  3. Click on any word to open the Study Mode popup
  4. Verify that the WBW translation in the word card displays in Urdu (not English)

  ### Edge Cases Verified

  - [x] ⏳ Loading state handled
  - [x] ❌ Error state handled
  - [x] 📭 Empty state handled
  - [x] 👤 Logged-in vs guest behavior (if applicable)

  ## Pre-Review Checklist

  ### Code Quality

  - [x] I have performed a **self-review** of my code (file by file)
  - [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
  - [x] No `any` types used (or justified if unavoidable)
  - [x] No unused code, imports, or dead code included
  - [x] Complex logic has inline comments explaining "why"
  - [x] Functions are under 30 lines and follow single responsibility

  ### Testing & Validation

  - [ ] All tests pass locally (`yarn test`)
  - [ ] Linting passes (`yarn lint`)
  - [ ] Build succeeds (`yarn build`)
  - [x] Edge cases and error scenarios are handled

  ### Documentation

  - [x] Code is self-documenting with clear naming
  - [x] README updated (if adding features or setup changes)
  - [x] Inline comments added for complex logic

  ### Localization (if UI changes)

  - [x] All user-facing text uses `next-translate`
  - [x] Only English locale files modified (Lokalise handles others)
  - [x] RTL layout verified

  ### Accessibility (if UI changes)

  - [x] Semantic HTML elements used
  - [x] ARIA attributes added where needed
  - [x] Keyboard navigation works

  ## AI Assistance Disclosure

  - [ ] AI tools were NOT used for this PR
  - [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code


[QF-4667]: https://quranfoundation.atlassian.net/browse/QF-4667?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/QuranReader/ReadingView/StudyModeModal/index.tsx [hunk 1]
```diff
@@ -157,7 +159,7 @@ const StudyModeModal: React.FC<Props> = ({
         translations: selectedTranslations.join(','),
         ...getDefaultWordFields(quranReaderStyles.quranFont),
         ...getMushafId(quranReaderStyles.quranFont, quranReaderStyles.mushafLines),
-        wordTranslationLanguage: 'en',
+        wordTranslationLanguage: wordByWordLocale,
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/QuranReader/ReadingView/StudyModeModal/index.tsx:162]: good catch will do it
